File: backend/internal/model/product.go
package model

import (
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Name        string  `json:"name" gorm:"not null"`
	Description string  `json:"description"`
	Price       float64 `json:"price" gorm:"not null"`
	Category    string  `json:"category" gorm:"index"`
	ImageURL    string  `json:"image_url"`
	Stock       int     `json:"stock" gorm:"default:0"`
	Status      string  `json:"status" gorm:"default:active"`
}

File: frontend/src/api/product.ts
import api from './index'

export interface Product {
  id: number
  name: string
  description: string
  price: number
  category: string
  image_url: string
  stock: number
  status: string
  created_at: string
}

export const productApi = {
  list(category?: string) {
    return api.get<Product[]>('/products', { params: { category } })
  },

  detail(id: number) {
    return api.get<Product>(`/products/${id}`)
  },
}
